tools/debugging/json-log-to-html.py

```diff
@@ -0,0 +1,202 @@
+"""
+Utility to format Raiden json logs into HTML.
+Colorizes log key-values according to their hash to make debugging easier.
+Allows to filter records by `event`.
+When processing multiple files make sure to set PYTHONHASHSEED to a fixed value,
+otherwise the colors of values will not match up.
+"""
+
+import hashlib
+import json
+from collections import Counter, namedtuple
+from datetime import datetime
+from html import escape
+from json import JSONDecodeError
+
+import click
+
+try:
+    from colour import Color
+except ImportError:
+    click.secho('This tool requires the `colour` package.')
+    import sys
+    sys.exit(1)
+
+
+Line = namedtuple('Line', ('event', 'timestamp', 'logger', 'level', 'fields'))
+
+TEMPLATE = """\
+<!doctype html>
+<html>
+<head>
+<style>
+* {{
+    font-family: Helvetica, sans-serif
+}}
+body {{
+    background: #202020;
+    color: white;
+}}
+table {{
+    white-space: nowrap;
+}}
+table tr.head {{
+    position: sticky;
+}}
+table tr:nth-child(2) {{
+    padding-top: 15px;
+}}
+table tr:nth-child(odd) {{
+    background-color: #303030
+}}
+table td {{
+    padding-right: 5px;
+}}
+.lvl-debug {{
+    color: #20d0d0;
+}}
+.lvl-info {{
+    color: #20d020;
+}}
+.lvl-warning {{
+    color: #d0d020;
+}}
+.lvl-error {{
+    color: #d04020;
+}}
+.fn {{
+    color: #f040f0;
+}}
+</style>
+<body>
+<h1>{name}</h1>
+<h2>Generated on: {date:%Y-%m-%d %H:%M}</h2>
+<table>
+{table_header}
+{table_rows}
+</table>
+</body>
+</html>
+"""
+
+
+def rgb_color_picker(obj, min_luminance=None, max_luminance=None):
+    """Modified version of colour.RGB_color_picker"""
+    color_value = int.from_bytes(
+        hashlib.md5(str(obj).encode('utf-8')).digest(),
+        'little',
+    ) % 0xffffff
+    color = Color(f'#{color_value:06x}')
+    if min_luminance and color.get_luminance() < min_luminance:
+        color.set_luminance(min_luminance)
+    elif max_luminance and color.get_luminance() > max_luminance:
+        color.set_luminance(max_luminance)
+    return color
+
+
+def parse_log(log_file):
+    known_fields = Counter()
+    log_records = []
+    for i, line in enumerate(log_file, start=1):
+        try:
+            line_dict = json.loads(line.strip())
+        except JSONDecodeError as ex:
+            click.secho(f'Error parsing line {i}: {ex}')
+            continue
+        log_records.append(
+            Line(
+                line_dict.pop('event'),
+                line_dict.pop('timestamp'),
+                line_dict.pop('logger'),
+                line_dict.pop('level'),
+                line_dict,
+            ),
+        )
+        for field_name in line_dict.keys():
+            known_fields[field_name] += 1
+
+    return log_records, known_fields
+
+
+def filter_records(log_records, drop_event):
+    return (
```

Comment:
it may also be beneficial to allow filtering by logger. Or else this happens.
![2018-12-03-142618_1546x523_scrot](https://user-images.githubusercontent.com/1658405/49376557-e2ff1600-f707-11e8-8ed7-433ef30566bb.png)

Reply:
Makes sense

Reply:
Also done